core/src/err/mod.rs

```diff
@@ -861,9 +861,17 @@ pub enum Error {
 	InvalidPass,
 
 	/// There was an error with authentication
+	///
+	/// This error hides different kinds of errors directly related to authentication
 	#[error("There was a problem with authentication")]
 	InvalidAuth,
 
+	/// There was an unexpected error while performing authentication
+	///
+	/// This error hides different kinds of unexpected errors that may affect authentication
+	#[error("There was an unexpected error while performing authentication")]
+	UnexpectedAuth,
```

Comment:
We need to add new errors to the end of the enum right? An older client will have the id for the error in the enum represented as a different error

Reply:
As we spoke, it is my understanding that this is not the case. Errors are not serialized by their position on the enum but rather by their string representation. If that is correct, it does not matter where the error is on the enum.